src/index/base.cpp

```diff
@@ -372,3 +372,14 @@ IndexSummary BaseIndex::GetSummary() const
     summary.best_block_height = m_best_block_index ? m_best_block_index.load()->nHeight : 0;
     return summary;
 }
+
+void BaseIndex::SetBestBlockIndex(const CBlockIndex* block) {
+    assert(!node::fPruneMode || AllowPrune());
+
+    m_best_block_index = block;
+    if (AllowPrune() && block) {
+        node::PruneLockInfo prune_lock;
+        prune_lock.m_height_first = block->nHeight;
+        WITH_LOCK(::cs_main, m_chainstate->m_blockman.UpdatePruneLock(GetName(), prune_lock));
```

Comment:
This updates the prune lock for coinstatsindex and blockfilterindex regularly, even if we don't have pruning enabled. Is that necessary?

Reply:
I was feeling the same and previously had a commit included here but removed it due to feedback that it wasn't needed: https://github.com/bitcoin/bitcoin/commit/e365e87a72cc49e175b31256a11a6c5b1621d1f2

I guess I could open a follow-up PR but for now I am trying to keep this PR as simple as possible to move it forward.